app/models/aaib_report.rb

```diff
@@ -0,0 +1,33 @@
+class AaibReport < Document
+
+  validates :date_of_occurrence, presence: true, date: true, unless: ->(report) {
+    report.report_type == "safety-study" && report.date_of_occurrence.blank?
+  }
+
+  FORMAT_SPECIFIC_FIELDS = [
+    :date_of_occurrence,
+    :aircraft_category,
+    :report_type,
+    :location,
+    :aircraft_type,
+    :registration,
+  ]
+
+  attr_accessor *FORMAT_SPECIFIC_FIELDS
+
+  def initialize(params = {})
+    super(params, FORMAT_SPECIFIC_FIELDS)
+  end
+
+  def format
+    "aaib_report"
+  end
+
+  def self.format
+    new.format
+  end
```

Comment:
Could this method implementation be pushed down to the `Document` class?  Are there expected to be any `Document` subclasses where the `<subclass>.format` is different to the `<subclass_instance>.format` ? If not I'd suggest flipping the implementation while you're at it (for boring micro performance optimization reasons - you'll have one copy of the format string per subclass, not one copy per instance):

```ruby
class Document
  def format
    self.class.format
  end

  def self.format
    raise NotImplementedError
  end
end

class FooBar < Document
  def self.format
    'foo_bar'
  end
end
```